projects/workflows-creator/src/lib/builder/builder.component.ts

```diff
@@ -462,8 +469,52 @@ export class BuilderComponent<E> implements OnInit, OnChanges {
    * It builds a new diagram, emits the new diagram, and then tells Angular to update the view
    */
   async updateDiagram() {
+    const nodes = [
+      ...this.eventGroups[0].children,
+      ...this.actionGroups[0].children,
+      ...this.elseActionGroups[0].children,
+    ];
+    let isValid =
+      !!this.eventGroups[0].children.length &&
+      (!!this.actionGroups[0].children.length ||
+        !!this.elseActionGroups[0].children.length);
+    if (isValid) {
+      for (const node of nodes) {
+        switch (node.node.getIdentifier()) {
+          case EventTypes.OnChangeEvent:
+          case EventTypes.OnValueEvent:
+          case ActionTypes.ChangeColumnValueAction:
+            const columnExists = !!node.node.state.get('column');
+            const valueExists = !!node.node.state.get('value');
+            const valueTypeIsAnyValue =
+              node.node.state.get('valueType') === ValueTypes.AnyValue;
+            isValid = columnExists && (valueExists || valueTypeIsAnyValue);
+            break;
```

Comment:
Shouldn't this be part of the respective node classes?

Reply:
yes it should be , can u plz tell me the approach which i should follow sir, but sir we need to give this in emergency so

Reply:
sure sir ,I will create the seprate issue of this .